Enterprise-CMCS/macpro-onemac
# PI2 26571: Setting Up Google Analytics (OneMAC)

Story: https://qmacbis.atlassian.net/browse/OY2-26571
Endpoint: See github-actions bot comment

### Details
The complete task is to migrate the Google Analytics from OneMAC to MAKO - but GA in OneMAC does not appear to be working.  These updates should get the GA tagging set per deployment environment using the Action Secrets to hold the values for the different branches.

### Changes
- updated the tag code from Google because old tags/code wasn't working (GA was updated last summer)
- Added the REACT_APP_GOOGLE_TAG parameter to the GitHub Action secrets
- updated the index.html to use the stored value for the tag

### Implementation Notes
- must put REACT_APP_ in front of environment variables to get used in index.html ?? not entirely sure why, but it didn't work without it and does work with it... so.... 

### Test Plan
1. Create a new branch from this branch
2. Deploy branch
3. Verify that the application's Google Tag is the default "notag"
4. Create an Action Secret with <branch_name>_REACT_APP_GOOGLE_TAG and "newtag" (suggested value)
5. re-deploy (might not need to?  maybe can just restart the application...)
6. Verify that the application's Google Tag is new value
